src/initialize_dae.jl

```diff
@@ -150,19 +174,26 @@ function _initialize_dae!(integrator, prob::ODEProblem, alg::ShampineCollocation
             out .-= tmp
             nothing
         end
-        r = nlsolve(nlequation!, u0, autodiff = isad ? :forward : :central,
-                    method = :newton)
-        integrator.u .= r.zero
+
+        nlsolve = default_nlsolve(alg.nlsolve, isinplace, u0, isad)
+
+        nlfunc = NonlinearFunction(nlequation!; jac = f.jac,
```

Comment:
The Jacobian of this function is not the same as the Jacobian of the ODE. `f.jac` would have to be modified to be useful here. Instead it's `M/dt - J` for J being `f.jac` (I think, someone would need to double check. Shampine collocation init is simply an implicit Euler step that is used to converge the algebraic variables, so it should be that, one would just have to double check the place of the constant dt), so one would need to `if has_jac(f); new_jac(du,u,p,) = ...` and define that equation. Let's leave that for a follow up PR though.